lib/index.js

```diff
@@ -58,20 +57,31 @@ Vero.prototype.group = request('/users/edit', 'put');
  * @api public
  */
 
-Vero.prototype.track = function(payload, fn) {
+Vero.prototype.track = function(track, fn) {
+  var payload = mapper.track.call(this, track);
   var endpoint;
   // event name isn't included in unsubscribe calls
   if (payload.event_name) {
     endpoint = '/events/track';
   } else {
     endpoint = '/users/unsubscribe';
   }
+  var self = this;
   return this
     .post(endpoint)
     .set('Accept', 'application/json')
     .type('json')
     .send(payload, this.settings)
-    .end(this.handle(fn));
+
+    // Tags are sent after the initial event to ensure the existence of the object to be tagged in Vero.
+    // On error, the callback is invoked.
+    // See docs: https://visionmedia.github.io/superagent/#request-basics
+
+    .then(function() {
+      var tags = checkForTags(track);
+      if (tags) return self.addOrRemoveTags(track, tags, fn);
+      return fn();
```

Comment:
@ccnixon i think you want to pass something here. if there are no tags we should expose the `res` of the initial request. 

@tsholmes do you think this scenario requires the need to use `Batch`? If the `tags` exists, that would be the proper way to expose both responses correct?